app/src/main/java/net/squanchy/proximity/preconditions/OptInPreferencePersister.java

```diff
@@ -0,0 +1,21 @@
+package net.squanchy.proximity.preconditions;
+
+import android.content.SharedPreferences;
+
+public class OptInPreferencePersister {
+
+    private static final String OPT_IN_PREFERENCE_KEY = "opt_in_preference";
+    private final SharedPreferences preferences;
+
+    public OptInPreferencePersister(SharedPreferences preferences) {
+        this.preferences = preferences;
+    }
+
+    public void setOptInPreferenceTo(boolean granted) {
```

Comment:
I would rather have two public methods for the two cases, with more descriptive names, such as: `storeUserOptedIn()` e `storeUserOptedOut()`?

Using a boolean flag as the only parameter in APIs is a discouraged practice. The [alternative](https://books.google.co.uk/books?id=ka2VUBqHiWkC&lpg=PA190&ots=yZLgJjr_U-&pg=PA190#v=onepage&q&f=false) would be using an enum with two values (e.g., `storeUserPreference(OptIn optIn)` with two possible values, `OPTED_IN` and `OPTED_OUT`). 

I'd rather have the two methods though.